src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int sizeOfFilledList;
+
+    public MyLinkedList() {
+        this.head = null;
+        this.tail = null;
+        this.sizeOfFilledList = 0;
+    }
+
+    private static class Node<T> {
+        private Node<T> previous;
+        private T value;
+        private Node<T> next;
+
+        public Node(Node<T> previous, T value, Node<T> next) {
+            this.previous = previous;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        linkTail(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        if (index == sizeOfFilledList) {
+            add(value);
+        } else if (index < sizeOfFilledList && index >= 0) {
+            linkNode(value, findNodeByIndex(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findNodeByIndex(index);
+        T currentNodeValue = currentNode.value;
+        currentNode.value = value;
+        return currentNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return removeNode(findNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (int i = 0; i < sizeOfFilledList; i++) {
+            T valueToRemove = findNodeByIndex(i).value;
```

Comment:
why do we use the method findNodeByIndex() here?
let's use for loop for iteration